### PR TITLE
Add table aws_sagemaker_training_job. Closes #348

### LINKED ISSUE #348: Add table aws_sagemaker_training_job
**References**
https://docs.aws.amazon.com/sagemaker/latest/dg/train-model.html

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Terraform not supported. Resource creation is failed through cli  and also this resource doesn't have any delete api call
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  arn,
  training_job_status,
  creation_time,
  last_modified_time
from
  aws_sagemaker_training_job;

+-----------------------------+-----------------------------------------------------------------------------------+---------------------+---------------------+---------------------+
| name                        | arn                                                                               | training_job_status | creation_time       | last_modified_time  |
+-----------------------------+-----------------------------------------------------------------------------------+---------------------+---------------------+---------------------+
| jobtesting                  | arn:aws:sagemaker:us-east-1:986325076436:training-job/jobtesting                  | Stopped             | 2021-05-10 12:48:24 | 2021-05-10 12:50:40 |
| turbottest56375             | arn:aws:sagemaker:us-east-1:986325076436:training-job/turbottest56375             | Failed              | 2021-05-10 13:04:10 | 2021-05-10 13:05:30 |
| pca-2021-04-08-09-31-38-028 | arn:aws:sagemaker:us-east-1:986325076436:training-job/pca-2021-04-08-09-31-38-028 | Completed           | 2021-04-08 09:31:38 | 2021-04-08 09:36:02 |
+-----------------------------+-----------------------------------------------------------------------------------+---------------------+---------------------+---------------------+


select
  name,
  arn,
  resource_config ->> 'InstanceType' as instance_type,
  resource_config ->> 'InstanceCount' as instance_count,
  resource_config ->> 'VolumeKmsKeyId' as volume_kms_id,
  resource_config ->> 'VolumeSizeInGB' as volume_size
from
  aws_sagemaker_training_job;

+-----------------------------+-----------------------------------------------------------------------------------+---------------+----------------+---------------+-------------+
| name                        | arn                                                                               | instance_type | instance_count | volume_kms_id | volume_size |
+-----------------------------+-----------------------------------------------------------------------------------+---------------+----------------+---------------+-------------+
| pca-2021-04-08-09-31-38-028 | arn:aws:sagemaker:us-east-1:986325076436:training-job/pca-2021-04-08-09-31-38-028 | ml.c4.xlarge  | 1              | <null>        | 30          |
| turbottest56375             | arn:aws:sagemaker:us-east-1:986325076436:training-job/turbottest56375             | ml.c4.xlarge  | 1              | <null>        | 30          |
| jobtesting                  | arn:aws:sagemaker:us-east-1:986325076436:training-job/jobtesting                  | ml.c4.xlarge  | 1              | <null>        | 30          |
+-----------------------------+-----------------------------------------------------------------------------------+---------------+----------------+---------------+-------------+

select
  name,
  arn,
  training_job_status,
  failure_reason
from
  aws_sagemaker_training_job
where
  training_job_status = 'Failed';

+-----------------+-----------------------------------------------------------------------+---------------------+------------------------------------------------------------------------------
| name            | arn                                                                   | training_job_status | failure_reason                                                               
+-----------------+-----------------------------------------------------------------------+---------------------+------------------------------------------------------------------------------
| turbottest56375 | arn:aws:sagemaker:us-east-1:986325076436:training-job/turbottest56375 | Failed              | ClientError: SageMaker was unable to assume the role 'arn:aws:iam::9863250764
+-----------------+-----------------------------------------------------------------------+---------------------+------------------------------------------------------------------------------

select
  secondary_status,
  count(*)
from
  aws_sagemaker_training_job
group by
  secondary_status;

+------------------+-------+
| secondary_status | count |
+------------------+-------+
| Completed        | 1     |
| Failed           | 1     |
| Stopped          | 1     |
+------------------+-------+
```
</details>

### CODE REVIEW

File: aws/table aws_sagemaker_training_job.go [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+func tableAwsSageMakerTrainingJob(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_sagemaker_training_job",
+		Description: "AWS Sagemaker Training Job",
```
Comment (khushboo9024) [inline aws/table aws_sagemaker_training_job.go:16]:
```suggestion
		Description: "AWS SageMaker Training Job",
```

File: aws/table aws_sagemaker_training_job.go [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTION
+
+func sageMakerTrainingJobtagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table aws_sagemaker_training_job.go:393]:
```suggestion
func sageMakerTrainingJobTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```

File: aws/table aws_sagemaker_training_job.go [hunk 3]
```diff
@@ -0,0 +1,430 @@
+package aws
```
Comment (khushboo9024) [inline aws/table aws_sagemaker_training_job.go:1]: File name is wrong , it should be `table_aws_sagemaker_training_job.go`

File: aws/table aws_sagemaker_training_job.go [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+			{
+				Name:        "billable_time_in_seconds",
+				Description: "The billable time in seconds. Billable time refers to the absolute wall-clock time.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "checkpoint_config",
+				Description: "The billable time in seconds. Billable time refers to the absolute wall-clock time.",
+				Type:        proto.ColumnType_INT,
```
Comment (khushboo9024) [inline aws/table aws_sagemaker_training_job.go:59]:
```suggestion
				Type:        proto.ColumnType_JSON,
```

File: aws/table aws_sagemaker_training_job.go [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "enable_network_isolation",
+				Description: "Specifies enable network isolation for training jobs.",
+				Type:        proto.ColumnType_BOOL,
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "enableInter_container_traffic_encryption",
```
Comment (khushboo9024) [inline aws/table aws_sagemaker_training_job.go:80]:
```suggestion
				Name:        "enable_inter_container_traffic_encryption",
```

File: docs/tables/aws_sagemaker_training_job.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  aws_sagemaker_training_job
+where
+  training_job_status = 'Failed';
+```
+
+### List training jobs by status
+
+```sql
+select
+  secondary_status,
```
Comment (khushboo9024) [inline docs/tables/aws_sagemaker_training_job.md:52]: Is this query useful?

File: aws/table_aws_sagemaker_training_job.go [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the training job.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("TrainingJobName"),
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the training job.",
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_training_job.go:35]:
```suggestion
				Description: "The Amazon Resource Name (ARN) specifying the training job.",
```

File: aws/table_aws_sagemaker_training_job.go [hunk 8]
```diff
@@ -0,0 +1,430 @@
...
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var arn string
+	if h.Item != nil {
+		arn = trainingJobArn(h.Item)
+	} else {
+		arn = d.KeyColumnQuals["arn"].GetStringValue()
+	}
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_training_job.go:369]: Do we need this block for `hydrate` function?

File: aws/table_aws_sagemaker_training_job.go [hunk 9]
```diff
@@ -0,0 +1,430 @@
...
+	// Get call
+	op, err := svc.ListTags(params)
+	if err != nil {
+		return nil, err
+	}
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTION
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_training_job.go:391]:
```suggestion
//// TRANSFORM FUNCTIONS
```

File: aws/table_aws_sagemaker_training_job.go [hunk 10]
```diff
@@ -0,0 +1,430 @@
...
+func getAwsSageMakerTrainingJob(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var name string
+	if h.Item != nil {
+		name = trainingJobName(h.Item)
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_training_job.go:329]: Do we require to check output `type` for `GET` function?
I think the type will be same as `LIST` i.e. `*sagemaker.TrainingJobSummary`

File: aws/table_aws_sagemaker_training_job.go [hunk 11]
```diff
@@ -0,0 +1,430 @@
...
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the training job.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJobTags,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns for all tables
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_training_job.go:265]:
```suggestion
			// Steampipe standard columns
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by khushboo9024 — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**